trin-core/src/portalnet/overlay_service.rs

```diff
@@ -1264,7 +1269,12 @@ where
     }
 
     // Process ACCEPT response
-    fn process_accept(&self, response: Accept, enr: Enr, offer: Request) -> anyhow::Result<Accept> {
+    fn process_accept(
+        &mut self,
```

Comment:
It doesn't seem to me like this update to a mutable borrow is necessary - am I missing something?

Reply:
yea, the `update_node_radius` requires a mutable `self` because it uses `self`'s `kbuckets` property mutably

Reply:
Good point, I think I had needed it previously because of something I was trying but ditched.

But this also made me realize that `process_pong` and `process_ping` are both borrowing `self` mutably when they don't need to be. Weird that the rust compiler doesn't complain about that.